src/rdiff.c

```diff
@@ -62,6 +62,9 @@
 #include "librsync.h"
 #include "isprefix.h"
 
+/** Whole file IO buffer sizes, from whole.c */
```

Comment:
Err... we don't want to do this. The whole point of setting the rs_inbuflen and rs_outbuflen defined in whole.c and exported from the rsync dynamically linked library in librsync.h is those are the variables used by whole.c for the input and output buffer sizes.

If we define and set our own inbuflen and outbuflen here in rdiff.c, it will have no effect on the buffer sizes used by whole.c in the librsync library.

I wondered if this was a limitation of DLL's... perhaps it's not possible to export variables from a dll, only functions? However a quick search seems to indicate that it is possible;

https://stackoverflow.com/questions/3097548/exporting-global-variables-from-dll

There must be something else going wrong if this is not working.

Reply:
Actually, this raises questions about rs_librsync_version, which is currently declared in librsync.h, but is missing the LIBRSYNC_EXPORT qualifier there, and is defined in version.c with LIBRSYNC_EXPORT, and is used in rdiff.c.

How come you are not getting errors related to rs_librsync_version? I would expect that we need to add LIBRSYNC_EXPORT to it in librsync.h too.

Reply:
I can confirm that `rs_inbuflen` and `rs_outbuflen` cannot be initialized in `rdiff.c` due to [limitations](https://docs.microsoft.com/en-us/cpp/c-language/rules-and-limitations-for-dllimport-dllexport) of the `dllimport` attribute.

Reply:
I've decided to mark this as "[WIP]" until I can fix the `rs_inbuflen`/`rs_outbuflen` `initializer element is not constant` issue, because everything else seems to be resolved.